daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/DataLoc.scala

```diff
@@ -49,9 +49,9 @@ class DataLoc(
   Assert.usage(bitLimit1b.isEmpty || bitLimit1b.get >= 0)
   Assert.usage(bitPos1b >= 1)
 
-  val bitPos0b = math.max(bitPos1b - 1, 0).toInt
+  val bitPos0b = bitPos1b - 1
```

Comment:
I think it is worth it to add the type declaration of Long to these members. Especially public ones.